### PR TITLE
feat/govAnalytics

### PR DESCRIPTION
**Pull request by VAIBHAV-25**
## Summary by CodeRabbit

- **New Features**
	- Introduced an API endpoint for retrieving track-level voting analytics.
	- Implemented a server-side rendered page for governance level analytics, improving user experience.
	- Added "Gov Analytics" menu item to the application layout for easier navigation.
	- Expanded querying capabilities with new GraphQL queries for proposal tracking and status reporting.

- **Bug Fixes**
	- Enhanced error handling in the new API endpoint for better user feedback.

- **Documentation**
	- Updated documentation to reflect new API endpoints and analytics features.

- **Style**
	- Improved styling for the new analytics menu item and related components.

### CODE REVIEW

File: pages/api/v1/govAnalytics/statuswiseRefOutcome.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+		const subsquidRes = await fetchSubsquid({
+			network: network,
+			query: GET_STATUS_WISE_REF_OUTCOME,
+			variables: variables
+		});
+		const rawData = subsquidRes?.data || {};
+
+		const transformedData: Record<string, number> = Object.keys(rawData).reduce(
+			(acc, key) => {
+				acc[key] = rawData[key].totalCount;
```
Comment (KanishkaRajputd) [inline pages/api/v1/govAnalytics/statuswiseRefOutcome.ts:39]: put null check
incase key or value undefined

File: pages/gov-level-analytics/index.tsx [hunk 2]
```diff
@@ -0,0 +1,57 @@
+// Copyright 2019-2025 @polkassembly/polkassembly authors & contributors
```
Comment (KanishkaRajputd) [inline pages/gov-level-analytics/index.tsx:1]: route name should be analytics

File: src/components/AppLayout/index.tsx [hunk 3]
```diff
@@ -42,7 +42,8 @@ import {
 	ApplayoutIdentityIcon,
 	ArchivedIcon,
 	ClearIdentityOutlinedIcon,
-	RoundedDollarIcon
+	RoundedDollarIcon,
```
Comment (KanishkaRajputd) [inline src/components/AppLayout/index.tsx:45]: don't import icon from custom icon and don't initialize there 
use <Image and dark-icons class name for dark mode

File: src/components/GovAnalytics/AnalyticsDelegationTrends.tsx [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+		}
+	};
+
+	useEffect(() => {
+		getData();
+		// eslint-disable-next-line react-hooks/exhaustive-deps
+	}, []);
+
+	// eslint-disable-next-line @typescript-eslint/no-unused-vars
+	const [noData, setNoData] = useState<boolean>(false);
```
Comment (KanishkaRajputd) [inline src/components/GovAnalytics/AnalyticsDelegationTrends.tsx:66]: no need set a state for no data just check with !loading and !data?.length

File: src/components/GovAnalytics/AnalyticsReferendumOutcome.tsx [hunk 5]
```diff
@@ -0,0 +1,252 @@
...
+	const getData = async () => {
+		setLoading(true);
+		try {
+			const { data } = await nextApiClientFetch<IGetStatusWiseRefOutcome>('/api/v1/govAnalytics/statuswiseRefOutcome', {
+				trackId: selectedTrack === null ? null : trackIds[selectedTrack]
+			});
+			if (data) {
+				setStatusInfo(data.statusCounts);
+				setLoading(false);
+			}
```
Comment (KanishkaRajputd) [inline src/components/GovAnalytics/AnalyticsReferendumOutcome.tsx:71]: no need for try catch here just check error and data coming from api

File: src/components/GovAnalytics/AnalyticsReferendumCount.tsx [hunk 6]
```diff
@@ -0,0 +1,202 @@
...
+		setLoading(true);
+		try {
+			const { data } = await nextApiClientFetch<IGetStatusWiseProposalCount>('/api/v1/govAnalytics/categoryWiseTotalProposalCount', {
+				categoryIds: groupedTrackIds
+			});
+			if (data) {
+				setCategoryInfo(data?.categoryCounts);
+				setLoading(false);
+			}
+		} catch (error) {
```
Comment (KanishkaRajputd) [inline src/components/GovAnalytics/AnalyticsReferendumCount.tsx:76]: no need for try catch here just check error and data coming from api

File: src/components/GovAnalytics/AnalyticsStats.tsx [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+	const [totalApprovedProposalCount, setTotalApprovedProposalCount] = useState(0);
+	const [loading, setLoading] = useState<boolean>(false);
+
+	const getData = async () => {
+		setLoading(true);
+		const url = '/api/v1/govAnalytics/totalApprovedProposalCount';
+		try {
+			const { data } = await nextApiClientFetch<IGetTotalApprovedProposalCount>(url);
+			if (data) {
+				setTotalApprovedProposalCount(data?.totalCount);
```
Comment (KanishkaRajputd) [inline src/components/GovAnalytics/AnalyticsStats.tsx:26]: no need for try catch here just check error and data coming from api

File: src/components/GovAnalytics/AnalyticsStats.tsx [hunk 8]
```diff
@@ -0,0 +1,108 @@
...
+					</div>
+				</div>
+
+				<Divider
+					className='hidden h-[87px] bg-section-light-container dark:bg-separatorDark sm:flex'
+					type='vertical'
+				/>
+				<Divider className=' bg-section-light-container dark:bg-separatorDark sm:hidden' />
+
+				{/* Total Voting Power */}
```
Comment (KanishkaRajputd) [inline src/components/GovAnalytics/AnalyticsStats.tsx:63]: remove comments

File: src/components/GovAnalytics/GovAnalyticsConvictions.tsx [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+const AnalyticsDelegationSplitGraph = dynamic(() => import('../TrackLevelAnalytics/AnalyticsVotingTrends/TrackAnalyticsgraphs/AnalyticsDelegationSplitGraph'), { ssr: false });
+const AnalyticsVoteSplitGraph = dynamic(() => import('../TrackLevelAnalytics/AnalyticsVotingTrends/TrackAnalyticsgraphs/AnalyticsVoteSplitGraph'), { ssr: false });
+
+const GovAnalyticsConvictions = ({ isSmallScreen }: { isSmallScreen: boolean }) => {
+	const { votes } = useTrackLevelAnalytics();
+
+	const convictionVotesData = getAnalyticsVotesByFilter(votes, ETrackLevelAnalyticsFilterBy.CONVICTION_VOTES);
+
+	const delegationSplit = convictionVotesData.sort((a, b) => a.delegationSplitData.index - b.delegationSplitData.index).map((item) => item.delegationSplitData);
+	const votesSplit = convictionVotesData.sort((a, b) => a.votesSplitData.index - b.votesSplitData.index).map((item) => item.votesSplitData);
```
[the remark this answers is not in the document]
Comment (KanishkaRajputd) [reply src/components/GovAnalytics/GovAnalyticsConvictions.tsx:19]: check this

File: src/components/AppLayout/index.tsx (hunk 3 above)
Comment (VAIBHAV-25) [reply src/components/AppLayout/index.tsx:45]: tried but it was breaking in dark mode

File: pages/api/v1/govAnalytics/allTracksAnalytics.ts [hunk 10]
```diff
@@ -0,0 +1,71 @@
...
+			let totalVotes = 0;
+
+			const trackSnapshot = await networkDocRef(network).collection('track_level_analytics').doc(String(trackNumber))?.collection('votes').get();
+
+			if (trackSnapshot.empty) return res.status(404).json({ message: 'No data found' });
+
+			trackSnapshot.docs.map((doc) => {
+				const data = doc.data();
+				if (!data?.voteAmount?.supportData) {
+					return res.status(404).json({ message: 'No data found' });
```
Comment (KanishkaRajputd) [inline pages/api/v1/govAnalytics/allTracksAnalytics.ts:34]: don't return from here just consider data value as 0

Comment (KanishkaRajputd) [reply pages/api/v1/govAnalytics/allTracksAnalytics.ts:34]: like if (data){
const supportPercentage = data.voteAmount.supportData.percentage || 0;
				const roundedNumber = Number(supportPercentage).toFixed(2);
				totalSupportedVoted += parseFloat(roundedNumber);
				totalVotes += 1;}

**Review by KanishkaRajputd — APPROVED** [review]

**Pull request merged**